libcloud/storage/drivers/azure_blobs.py

```diff
@@ -156,6 +156,27 @@ class AzureBlobsConnection(AzureConnection):
     """
     Represents a single connection to Azure Blobs
     """
+    def __init__(self, *args, **kwargs):
+        self.account_prefix = kwargs.pop('account_prefix', None)
+        super(AzureBlobsConnection, self).__init__(*args, **kwargs)
+
+    def morph_action_hook(self, action):
+        action = super(AzureBlobsConnection, self).morph_action_hook(action)
+
+        # The main Azure Blob Storage service uses the hostname to distinguish
+        # between accounts, e.g. `theaccount.blob.core.windows.net`.
+        # However, some custom deployments of the service such as the official
+        # emulator (https://github.com/Azure/Azurite) instead use a URL prefix
+        # such as `/theaccount`. The use-case is drivers instantiated via
+        # `host=somewhere.tld` and `key='theaccount'`.
+        # If a user wants to specify a custom host without an account prefix,
+        # e.g. for use-cases where the custom host implements an auditing
+        # proxy or similar, they can instantiate the driver with
+        # `host=theaccount.somewhere.tld` and `key=''`.
+        if self.account_prefix:
```

Comment:
consider "is not None", as [per the contributing guide](https://libcloud.readthedocs.io/en/latest/development.html#prefer-to-use-is-not-none-when-checking-if-a-variable-is-provided-or-defined)

Reply:
Done.